samples/extensions/extended_dynamic_state2/extended_dynamic_state2.cpp

```diff
@@ -0,0 +1,1065 @@
+
+	rasterization_state.frontFace = VK_FRONT_FACE_COUNTER_CLOCKWISE;
+
+	shader_stages[0] = load_shader("extended_dynamic_state2/background.vert", VK_SHADER_STAGE_VERTEX_BIT);
+	shader_stages[1] = load_shader("extended_dynamic_state2/background.frag", VK_SHADER_STAGE_FRAGMENT_BIT);
+
+	VK_CHECK(vkCreateGraphicsPipelines(get_device().get_handle(), pipeline_cache, 1, &graphics_create, VK_NULL_HANDLE, &pipeline.background));
+
+	/* Setup for third pipeline */
+	graphics_create.pTessellationState = &tessellation_state;
+	graphics_create.layout             = pipeline_layouts.tesselation;
+	input_assembly_state.topology      = VK_PRIMITIVE_TOPOLOGY_PATCH_LIST;
+
+	dynamic_state_enables.push_back(VK_DYNAMIC_STATE_PATCH_CONTROL_POINTS_EXT);
+	dynamic_state.pDynamicStates    = dynamic_state_enables.data();
+	dynamic_state.dynamicStateCount = static_cast<uint32_t>(dynamic_state_enables.size());
+
+	vertex_input_state.vertexBindingDescriptionCount   = static_cast<uint32_t>(vertex_input_bindings.size());
+	vertex_input_state.pVertexBindingDescriptions      = vertex_input_bindings.data();
+	vertex_input_state.vertexAttributeDescriptionCount = static_cast<uint32_t>(vertex_input_attributes.size());
+	vertex_input_state.pVertexAttributeDescriptions    = vertex_input_attributes.data();
+
+	/* Wireframe mode */
+	if (get_device().get_gpu().get_features().fillModeNonSolid)
+	{
+		rasterization_state.polygonMode = VK_POLYGON_MODE_LINE;
+	}
+
+	shader_stages[0]           = load_shader("extended_dynamic_state2/tess.vert", VK_SHADER_STAGE_VERTEX_BIT);
+	shader_stages[1]           = load_shader("extended_dynamic_state2/tess.frag", VK_SHADER_STAGE_FRAGMENT_BIT);
+	shader_stages[2]           = load_shader("extended_dynamic_state2/tess.tesc", VK_SHADER_STAGE_TESSELLATION_CONTROL_BIT);
+	shader_stages[3]           = load_shader("extended_dynamic_state2/tess.tese", VK_SHADER_STAGE_TESSELLATION_EVALUATION_BIT);
+	graphics_create.stageCount = static_cast<uint32_t>(shader_stages.size());
+	graphics_create.pStages    = shader_stages.data();
+	/* Enable depth test and write */
+	depth_stencil_state.depthWriteEnable = VK_TRUE;
+	depth_stencil_state.depthTestEnable  = VK_TRUE;
+	/* Flip cull mode */
+	rasterization_state.cullMode = VK_CULL_MODE_FRONT_BIT;
+	VK_CHECK(vkCreateGraphicsPipelines(get_device().get_handle(), pipeline_cache, 1, &graphics_create, VK_NULL_HANDLE, &pipeline.tesselation));
+}
+
+/**
+ * 	@fn void ExtendedDynamicState2::build_command_buffers()
+ * 	@brief Creating command buffers and drawing particular elements on window.
+ * 	@details Drawing object list:
+ * 			 - models from baseline scene
+ * 			 - model from tessellation scene
+ * 			 - background model
+ * 			 - primitive restart model
+ */
+void ExtendedDynamicState2::build_command_buffers()
+{
+	std::array<VkClearValue, 2> clear_values{};
+	clear_values[0].color        = {{0.0f, 0.0f, 0.0f, 0.0f}};
+	clear_values[1].depthStencil = {0.0f, 0};
+
+	int i = -1;
```

Comment:
Any reason for not just using a loop over i instead of the auto loop like the other samples do?

Reply:
I already discussed with author and change is required here.
Range-based loop was used to access element in vector, 'i' variable is only needed for framebuffers[i].
But in loop body instead of draw_cmd_buffer the element of vector is accessed by draw_cmd_buffers[i] -  this needs to be fixed.

Reply:
Changes merged